File: app/schemas/ship_booking_fee.py
# File: app/schemas/ship_booking_fee.py

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import FlightBookingFeeType


class ShipBookingFeeSettingOut(BaseModel):
    """تمثيل إعداد رسم حجز تذاكر البواخر الحالي في الاستجابات."""

    model_config = ConfigDict(from_attributes=True)

    fee_type: FlightBookingFeeType
    fee_value: Decimal
    updated_at: datetime | None


class ShipBookingFeeUpdateRequest(BaseModel):
    """طلب تحديث رسم حجز تذاكر البواخر (admin فقط)."""

    fee_type: FlightBookingFeeType
    fee_value: Decimal = Field(ge=0)


class ShipRouteQuoteOut(BaseModel):
    """
    تفصيل سعر حجز باخرة لعدد مسافرين مُحدَّد: السعر الحقيقي (يبقى كما
    ضبطه المدير دائماً)، رسم الحجز قبل وبعد أي خصم ساري، والسعر الإجمالي
    النهائي.
    """

    base_subtotal_usd: Decimal
    fee_amount_usd: Decimal
    discount_percentage: Decimal | None
    fee_after_discount_usd: Decimal
    total_price_usd: Decimal

File: frontend/src/types/shipBookingFee.ts
// File: frontend/src/types/shipBookingFee.ts

import type { FlightBookingFeeType } from "@/types/enums";

/** مطابق لـ app.schemas.ship_booking_fee.ShipBookingFeeSettingOut. */
export interface ShipBookingFeeSettingOut {
  fee_type: FlightBookingFeeType;
  fee_value: string;
  updated_at: string | null;
}

/** مطابق لـ app.schemas.ship_booking_fee.ShipBookingFeeUpdateRequest. */
export interface ShipBookingFeeUpdateRequest {
  fee_type: FlightBookingFeeType;
  fee_value: string;
}

/** مطابق لـ app.schemas.ship_booking_fee.ShipRouteQuoteOut. */
export interface ShipRouteQuoteOut {
  base_subtotal_usd: string;
  fee_amount_usd: string;
  discount_percentage: string | null;
  fee_after_discount_usd: string;
  total_price_usd: string;
}
